src/main/scala/nl/knaw/dans/lib/encode/package.scala

```diff
@@ -0,0 +1,66 @@
+/**
+ * Copyright (C) 2016 DANS - Data Archiving and Networked Services (info@dans.knaw.nl)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package nl.knaw.dans.lib
+
+import com.google.common.net.PercentEscaper
+import java.nio.file.Path
+import scala.collection.JavaConverters._
+
+package object encode {
+
+  // PercentEscaper that escapes all characters except '_' and English alphanumerical characters
+  private val percentEscaper = new PercentEscaper("_", false)
+
+  implicit class PathEncoding(val path: Path) extends AnyVal {
+
+    /**
+     * Escapes all characters in path segments, except those declared as "safe characters"
+     *
+     * @example
+     * {{{
+     *    import nl.knaw.dans.lib.encode._
+     *    import java.nio.file.Paths
+     *
+     *    val path = Paths.get("an/example/path_123/test-file(#3).txt")
+     *    val escapedPath = path.escapePath
+     *    // escapedPath: "an/example/path_123/test%2Dfile%28%233%29%2Etxt"
+     *  }}}
+     * @return an escaped path
+     */
+    def escapePath: String = {
+      path.asScala.map(_.toString).map(percentEscaper.escape).mkString("/")
```

Comment:
it would be more efficient to merge these two `map` calls

```suggestion
      path.asScala.map(_.toString.escapeString).mkString("/")
```

Reply:
A good point!